src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -73,6 +74,12 @@ export interface DiscoverSidebarProps {
    * Callback function to select another index pattern
    */
   setIndexPattern: (id: string) => void;
+  /**
+   * Metric tracking function
+   * @param metricType
+   * @param eventName
+   */
+  trackUiMetric: (metricType: UiStatsMetricType, eventName: string | string[]) => void;
```

Comment:
Think, `trackUiMetric` should be optional
```suggestion
  trackUiMetric?: (metricType: UiStatsMetricType, eventName: string | string[]) => void;
```